google/cloud/aiplatform/models.py

```diff
@@ -2036,3 +2058,138 @@ def list(
             location=location,
             credentials=credentials,
         )
+
+    @base.optional_sync()
+    def export_model(
+        self,
+        export_format_id: str,
+        artifact_destination: Optional[str] = None,
+        image_destination: Optional[str] = None,
+        sync: bool = True,
+    ) -> None:
+        """Exports a trained, exportable, Model to a location specified by the user.
+        A Model is considered to be exportable if it has at least one `supported_export_formats`.
+        Either `artifact_destination` or `image_destination` must be provided.
+
+        Usage:
+            my_model.export(
+                export_format_id='tf-saved-model'
+                artifact_destination='gs://my-bucket/models/'
+            )
+
+            or
+
+            my_model.export(
+                export_format_id='custom-model'
+                image_destination='us-central1-docker.pkg.dev/projectId/repo/image'
+            )
+
+        Args:
+            export_format_id (str):
+                Required. The ID of the format in which the Model must be exported.
+                Each Model lists the export formats it supports, which can be
+                found using `Model.supported_export_formats`.
+                If no value is provided here, then the first from the list
+                of the Model's supported formats is used by default.
+            artifact_destination (str):
+                The Cloud Storage location where the Model artifact is to be
+                written to. Under the directory given as the destination a
+                new one with name
+                "``model-export-<model-display-name>-<timestamp-of-export-call>``",
+                where timestamp is in YYYY-MM-DDThh:mm:ss.sssZ ISO-8601
+                format, will be created. Inside, the Model and any of its
+                supporting files will be written. This field should only be
+                set when the ``exportableContent`` field of the
```

Comment:
I'm not sure what this means because `supported_export_formats` is a list of strings and doesn't seem to have an `exportableContent` field.

Reply:
Perhaps we should expose `_supported_export_formats` directly instead of `supported_export_formats` in order for the user to see the format + content type pairings for themselves. Alternatively I can delete that portion of the docstrings for now.

Reply:
Is format + content required use this API effectively or is format enough?

Reply:
I think both are required to use the API effectively since you wouldn't know whether a particular export format can be saved as an image, artifact, or both. I'll go ahead and expose this in `supported_export_formats` and get rid of `_supported_export_formats` if there's no objection

Reply:
SGTM